apps/design/backend/src/test_decks.test.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  electionGeneralDefinition,
+  electionTwoPartyPrimaryDefinition,
+} from '@votingworks/fixtures';
+import {
+  BallotLayout,
+  DEFAULT_LAYOUT_OPTIONS,
+  layOutAllBallotStyles,
+} from '@votingworks/hmpb-layout';
+import {
+  BallotType,
+  ElectionDefinition,
+  getContests,
+} from '@votingworks/types';
+import { assert, iter } from '@votingworks/basics';
+import {
+  getBallotStyleById,
+  getBallotStylesByPrecinctId,
+  numBallotPositions,
+} from '@votingworks/utils';
+import { createPrecinctTestDeck } from './test_decks';
+
+function expectedTestDeckPages(
+  ballots: BallotLayout[],
+  electionDefinition: ElectionDefinition
+): number {
+  return iter(ballots)
+    .map((ballot) => {
+      const { document, gridLayout } = ballot;
+      const ballotStyle = getBallotStyleById(
+        electionDefinition,
+        gridLayout.ballotStyleId
+      );
+      const contests = getContests({
+        election: electionDefinition.election,
+        ballotStyle,
+      });
+      const maxContestOptions = Math.max(...contests.map(numBallotPositions));
+      const blankBallots = 2;
+      const overvotedBallots = 1;
+      return (
+        document.pages.length *
+        (maxContestOptions + blankBallots + overvotedBallots)
+      );
+    })
+    .sum();
+}
+
+// We test mainly that the test decks have the right number of pages, relying on
+// the fact that generateTestDeckBallots is tested in
+// libs/utils/src/test_deck_ballots.test.ts (ensuring we generate the ballots
+// with the proper votes) and that markBallot is tested by the ballot fixtures
+// in libs/hmpb/render-backend (ensuring the marks and write-ins look good) and
+// corresponding interpretation tests in libs/ballot-interpreter (ensuring the
+// marks and write-ins are interpreted correctly).
+//
+// Once ballot rendering is faster, it might be nice to also have a snapshot
+// test for test deck PDFs to ensure it all comes together correctly.
+describe('createPrecinctTestDeck', () => {
+  test('for a precinct with one ballot style', () => {
+    const electionDefinition = electionGeneralDefinition;
+    const { election } = electionDefinition;
+    const precinctId = election.precincts[0].id;
+    assert(
+      getBallotStylesByPrecinctId(electionDefinition, precinctId).length === 1
+    );
```

Comment:
Any reason not to use `expect(getBallotStylesByPrecinctId(electionDefinition, precinctId)).toHaveLength(1)`? It'd have a better error message on failure. There are other places this question applies too, such as `expect(foo.length).toEqual(bar)` instead of `expect(foo).toHaveLength(bar)`. Perhaps for consistency since you're doing other `.length`-based checks? Or perhaps because Jest would blow up trying to pretty-print the value passed to `expect`?

Reply:
I use `assert` when I'm trying to enforce an invariant on the test data. I use `assert` instead of `expect` to differentiate my intention - i.e. the test is invalid if this invariant breaks. But it seems like that wasn't so clear, so I'm open to suggestions on how to communicate that better.

I'll look into using `toHaveLength` more consistently for other cases.

Reply:
I did have an inkling of that rationale when reading it, so I don't think it was completely lost. I'm not really opposed to the way it is now–maybe include a comment? I don't feel strongly about it.